src/SearchPath.cc

```diff
@@ -29,27 +27,33 @@ static void add_applications_dir(std::string &str) {
     str += '/';
 }
 
-stringlist_t get_search_path() {
+stringlist_t build_search_path(std::string xdg_data_home, std::string home,
+                               std::string xdg_data_dirs,
+                               bool (*is_directory_func)(const std::string &)) {
```

Comment:
This could be a nitpick, but I'd use a template parameter instead of a function pointer. `build_search_path()` would have to be moved to `SearchPath.hh`.

Reply:
I don't have a strong opinion for either way, though using a template parameter also forces us to move `add_applications_dir` to the header, which then makes it visible to other translation units, so we then need to hide it (e.g. by nesting it inside `build_search_path`). So it's a little bit less straightforward than what I'd wish. Perhaps it'd be cleaner to encapsulate it into a class like in r2.18.

Reply:
OK. The current solution is good enough.